pkg/client/config/config.go

```diff
@@ -164,19 +159,32 @@ func NewRESTClient(config RestConfigClient) (*perseshttp.RESTClient, error) {
 		TLSClientConfig:     tlsCfg, // nolint: gas, gosec
 	}
 	var basicAuth *secret.BasicAuth
-
+	var httpClient *http.Client
 	if config.Auth != nil {
 		if config.Auth.BasicAuth != nil {
 			basicAuth = config.Auth.BasicAuth
 		}
 		if config.Auth.NativeAuth != nil {
 			roundTripper = transport.New(config.URL.URL, roundTripper, *config.Auth.NativeAuth)
 		}
+		if config.Auth.Oauth != nil {
+			oauthConfig := &clientcredentials.Config{
+				ClientID:     config.Auth.Oauth.ClientID,
+				ClientSecret: config.Auth.Oauth.ClientSecret,
+				TokenURL:     config.Auth.Oauth.TokenURL,
+				Scopes:       config.Auth.Oauth.Scopes,
+				AuthStyle:    config.Auth.Oauth.AuthStyle,
+			}
+			ctx := context.WithValue(context.Background(), oauth2.HTTPClient, roundTripper)
```

Comment:
I think this the http client that should go under this context value, not the transport

Reply:
oh yeah good call, thanks !

Reply:
done